pkg/ocm/billing.go

```diff
@@ -34,22 +36,63 @@ func (c *Client) GetBillingAccounts() ([]string, error) {
 		return nil, handleErr(quotaCostResponse.Error(), err)
 	}
 
-	var billingAccounts []string
+	var billingAccounts []*v1.CloudAccount
 	for _, item := range quotaCostResponse.Items().Slice() {
-		for _, account := range item.CloudAccounts() {
-			if account.CloudProviderID() == "aws" && !helper.Contains(billingAccounts, account.CloudAccountID()) {
-				billingAccounts = append(billingAccounts, account.CloudAccountID())
-			}
-		}
+		billingAccounts = append(billingAccounts, item.CloudAccounts()...)
 	}
 
 	if len(billingAccounts) == 0 {
-		return billingAccounts, errors.New("no billing accounts found")
+		return billingAccounts, errors.New("No billing account associated. " +
+			"Go to https://docs.openshift.com/rosa/rosa_architecture/rosa-understanding.html" +
+			" to learn how to associate your billing account." +
+			" You must associate a billing account to continue.")
 	}
 
 	return billingAccounts, nil
 }
 
+func GenerateBillingAccountsList(cloudAccounts []*v1.CloudAccount) []string {
+	var billingAccounts []string
+	for _, account := range cloudAccounts {
+		if account.CloudProviderID() == "aws" && !helper.ContainsPrefix(billingAccounts, account.CloudAccountID()) {
+			var contractString string
+			if HasValidContracts(account) {
+				contractString = " [Contract enabled]"
+			}
+			contractEnabledBillingAccount := account.CloudAccountID() + contractString
+			billingAccounts = append(billingAccounts, contractEnabledBillingAccount)
+		}
+	}
+	return billingAccounts
+
+}
+
+func GetNumsOfVCPUsAndClusters(dimensions []*v1.ContractDimension) (int, int) {
+	numOfVCPUs := 0
+	numOfClusters := 0
+	for _, dimension := range dimensions {
+		switch dimension.Name() {
+		case "four_vcpu_hour":
+			numOfVCPUs, _ = strconv.Atoi(dimension.Value())
+		case "control_plane":
```

Comment:
Just curious, but these dimensions are hardly available as consts from the SDK are they?

Reply:
Yep they are like this:
![image](https://github.com/openshift/rosa/assets/26282541/40fa0aa3-7e53-486a-9b1e-ef59c768300d)